### ISSUE TITLE
Add `type-literal` as an option for `vue/define-emits-declaration`

### ISSUE THREAD

**Issue by studnitz** [vuejs/eslint-plugin-vue#2242]
**What rule do you want to change?**

`vue/define-emits-declaration`

**Does this change cause the rule to produce more or fewer warnings?**

It's just a stylistic rule for enforcing the [more ergonomic `defineEmits` types](https://blog.vuejs.org/posts/vue-3-3#more-ergonomic-defineemits) introduced in Vue v3.3.

**How will the change be implemented? (New option, new default behavior, etc.)?**

Could be a new option `type-literal` alongside `type-based` or could be a configuration option when using `type-based`. One could also think about enforcing the usage of [labeled tuple elements](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html#labeled-tuple-elements) as shown in the example docs.

**Please provide some example code that this change will affect:**

```ts
// Valid with `type-literal`
const emit = defineEmits<{
  foo: [id: number]
  bar: [name: string, ...rest: any[]]
}>()

// Invalid with `type-literal`
const emit = defineEmits<{
  (e: 'foo', id: number): void
  (e: 'bar', name: string, ...rest: any[]): void
}>()
```

**What does the rule currently do for this code?**

Enforce either type-based or runtime based `defineEmits` declaration.

**What will the rule do after it's changed?**

Add also an option to enforce the new type syntax for `defineEmits` introduced in v3.3.

**Additional context**

**Comment by FloEdelmann** [comment]
A new `type-literal` option makes sense :+1: 

Is there any reason to not use labeled tuple elements? Otherwise, I'd suggest to always require them in the `type-literal` case. See also

* https://github.com/typescript-eslint/typescript-eslint/issues/2457

Happy to accept a PR for the new option!

**Comment by preetsahil** [comment]
Can i work on this issue ? or it is solved.

**Comment by FloEdelmann** [comment]
Feel free to work on it 👍
